src/Chapter3.hs

```diff
@@ -910,6 +1007,22 @@ Implement instances of "Append" for the following types:
 class Append a where
     append :: a -> a -> a
 
+newtype Gold = MkGold 
+  { unGold :: Integer
+  } deriving (Show)
+
+
+instance Append Gold where
+  append (MkGold x) (MkGold y) = MkGold (x + y)
+
+instance Append [a] where
+  append = (++)
+
+instance Append a => Append (Maybe a) where
+  append Nothing Nothing = Nothing
+  append Nothing y = y
+  append x Nothing = x
```

Comment:
This is an absolutely correct solution!

However, we can notice that these 3 cases could be reduced to 2!
Let's see what we can unite in here.

If some of the arguments are `Nothing`, you actually always return the second one! So you need to inspect only two cases: `Nothing` in the first position, and `Nothing` on the second one 🙂